### ISSUE TITLE
Partially broken keyboard navigation in newer versions of Poedit

### ISSUE THREAD

**Issue by k-kolev1985** [vslavik/poedit#415]
Since maybe v1.7 of Poedit, the change in the UI caused the keyboard navigation in Poedit to partially brake. I've tested now in v2.0.2 and I can navigate with TAB between the controls on the left (translations list, source, translation, etc.). But I seam not to be able to go to the controls located in the sidebar (suggestions, comments, notes, etc.). I can go to the comments and notes fields in v1.6 of Poedit, but not in newer versions of the program. Can this be fixed?

Thanks much in advance!

**Comment by vslavik** [comment]
Most of the content in the sidebar is *not* tab-navigatable — static text is static. Suggestions are accessible with Ctrl+<num> shortcuts.

The only element there that needs navigation is the *Add comment* button (and sometimes hyperlink at the top). And indeed, it would seem TAB navigation order is separate in left- and right-hand parts for those two, cycling independently.

**Comment by k-kolev1985** [comment]
In v1.6, the fields for comments and notes were focusable read-only fields, which screen readers can get to and read. If the current "static text" controls for comments and notes are not focusable, how can I get to them and read them with a screen reader? Executing the "Edit comment" command shows an empty field for entering a new comment - the current comment text is not shown there.

**Comment by vslavik** [comment]
> In v1.6, the fields for comments and notes were focusable read-only fields

I'm well aware of it, thank you. It didn't make sense to have it that way.

> If the current "static text" controls for comments and notes are not focusable, how can I get to them and read them with a screen reader?

How can you read the myriads of *labels* all over all the applications you have? Because screen readers don't need TAB navigation, they can see all the controls, not just the one that accept input (and so focus can be navigated to them).

> Executing the "Edit comment" command shows an empty field for entering a new comment - the current comment text is not shown there.

Edit → Edit comment of course does show the existing comment. You're either confusing comments with notes for translators or confusing Poedit with other software (e.g. NVDA's extras); I'm not sure which one is the case.

**Comment by k-kolev1985** [comment]
Yes, I'll try the other methods of reading the screen to get to the notes for translators.
And yes - I was confusing the comments and the notes for translators. Sorry about that.

**Comment by bdorer** [comment]
> > In v1.6, the fields for comments and notes were focusable read-only fields
> I'm well aware of it, thank you. It didn't make sense to have it that way.

Why doesn't make this sense? focusable read-only edit fields are good to read for all screenreaders. With NVDA I have to use object navigation and with JAWS I have to find the position with the mouse alias JAWS-Cursor.

> > If the current "static text" controls for comments and notes are not focusable, how can I get to them and read them with a screen reader?
> How can you read the myriads of _labels_ all over all the applications you have? Because screen readers don't need TAB navigation, they can see all the controls, not just the one that accept input (and so focus can be navigated to them).

that's true but you have to find a solution for every programme. Some lables are at the left, some at the top and so on. That's the difference between usable and accessible. Some software is accessible but not really usable because you have to use tab to go down a list. In Windows you use typically F6 to switch between areas of diffrent use. See Windows Explorer for an example.

Furthermore Poedit isn't using the same tab order everywhere. If I klick the quallity button and press escape, I can tab to add comment and cycle between this button.

**Comment by pvagner** [comment]
Hello,

Several pointers to this old issue. Unfortunatelly it still persists perhaps it has not been clearly explained.

* Some kind of navigation is possible for example clicking add comment button with the mouse, then pressing ok button in the edit comment dialog places focus to the edit comment button. Using tab or shift+tab is cycling in that part of the window, there is a link used to fetch online suggestions and some unlabelled button that I don't know what it's supposed to do. In order to get back to the list I have to click with the mouse. I can't figure out how to do that with the keyboard alone. Is that doable? Can you please reconsider that we need to be able to navigate there?
* Yes static text is static and it's being correctly used all over the place to label interactive controls such as edit fields, comboboxes and similar. Suggestions should not be presented as static text with the mouse click as the only action. Instead suggestions should be presented in a focusable and keyboard navigable list in order to make these really accessible. Yes we do have ability to review the whole screen or part of the screen but when handling this part of the app this way it's not very effective and we are loosing control relations. Can you please try to reconsider this? We need better access to suggestions for the future.
* The same goes for translator comments we can review the screen but we need this comment readily available. Either make it a read only text field as requested or put it into accessible description for some focusable control so we can quickly get to it when needed.

**Comment by vslavik** [comment]
> Instead suggestions should be presented in a focusable and keyboard navigable list in order to make these really accessible

They are listed in the suggestions menu.
